include/envoy/server/configuration.h

```diff
@@ -23,9 +23,9 @@ class Listener {
   virtual Network::FilterChainFactory& filterChainFactory() PURE;
 
   /**
-   * @return uint64_t the port.
+   * @return std::string the address.
    */
-  virtual uint64_t port() PURE;
+  virtual const std::string& address() PURE;
```

Comment:
@wattli this is where you need to return Address::InstancePtr. Resolve it inside the configuration. Otherwise https://github.com/lyft/envoy/blob/master/test/example_configs_test.cc#L32 will still potentially pass, since you first resolve it when the server starts running.